tests/integration_python/test_run_cli.py

```diff
@@ -372,3 +379,365 @@ def test_run_dry_run(pixi: Path, tmp_pixi_workspace: Path) -> None:
         stdout_excludes="WET",
         stderr_excludes="WET",
     )
+
+
+def test_run_args(pixi: Path, tmp_pixi_workspace: Path) -> None:
+    manifest = tmp_pixi_workspace.joinpath("pixi.toml")
+    toml = f"""
+    {EMPTY_BOILERPLATE_PROJECT}
+    [tasks]
+    """
+    manifest.write_text(toml)
+
+
+def test_invalid_task_args(pixi: Path, tmp_pixi_workspace: Path) -> None:
+    manifest = tmp_pixi_workspace.joinpath("pixi.toml")
+
+    config = tomli.loads(EMPTY_BOILERPLATE_PROJECT)
+
+    config["tasks"] = {
+        "task_invalid_defaults": {
+            "cmd": "echo Invalid defaults: {{ arg1 }} {{ arg2 }} {{ arg3 }}",
+            "args": [
+                {"name": "arg1", "default": "default1"},
+                "arg2",
+                {"name": "arg3", "default": "default3"},
+            ],
+        }
+    }
+
+    manifest.write_text(tomli_w.dumps(config))
+
+    verify_cli_command(
+        [
+            pixi,
+            "run",
+            "--manifest-path",
+            manifest,
+            "task_invalid_defaults",
+            "arg1",
+            "arg2",
+            "arg3",
+        ],
+        ExitCode.FAILURE,
+        stderr_contains="expected default value required after previous arguments with defaults",
+    )
+
+
+def test_task_args_with_defaults(pixi: Path, tmp_pixi_workspace: Path) -> None:
+    """Test tasks with all default arguments."""
+    manifest = tmp_pixi_workspace.joinpath("pixi.toml")
+
+    config = tomli.loads(EMPTY_BOILERPLATE_PROJECT)
```

Comment:
```suggestion
    manifest_path = tmp_pixi_workspace.joinpath("pixi.toml")

    manifest_content = tomli.loads(EMPTY_BOILERPLATE_PROJECT)
```

Pixi config is something else, so this would be confusing :)

Reply:
fixed this!